cookbook/networking/fetch-data.md

```diff
@@ -0,0 +1,282 @@
+---
+layout: page
+title: "Fetch data from the internet"
+---
+
+Fetching data from the internet is necessary for most apps. Luckily, Dart and Flutter provide convenient tools to work with data from the internet!
```

Comment:
I haven't read a Cookbook in a while, but this seems pretty course grained. Did we consider and evaluate smaller-sized recipes?

Reply:
Hey, good Q! Definitely trying to see if this is too much for one recipe. I can easily cut out the `FutureBuilder` section and focus only on making HTTP requests / parsing it. Would cut down some of the content, but I wanted to relate it back to the Flutter. Might have been a mistake...

Reply:
I assume you've seen what other cookbooks have done? It's been a while, but we can dig up some popular published cookbooks to see what worked there.

I'm personally a fan of smaller, composable recipes (instead of large, "5-course meals" :)

Reply:
This is the first pass at the cookbook; we can iterate on this, and I'm more than happy to organize a chat to talk about where to go next

Reply:
Yep! I checked out a couple of other Cookbooks, and this recipe didn't strike me as particularly long. For example, the React Native cookbook has several recipes which are a 3-5 pages long: https://books.google.de/books/about/React_Native_Cookbook.html?id=gs_cDgAAQBAJ&printsec=frontcover&source=kp_read_button&redir_esc=y#v=onepage&q&f=false

That said, I could easily see an argument for this particular recipe being shorter and more focused.